mountpoint-s3/src/prefetch/part_stream.rs

```diff
@@ -24,11 +26,12 @@ pub trait ObjectPartStream {
     /// size for the parts, but implementations are allowed to ignore it.
     fn spawn_get_object_request<Client>(
         &self,
-        client: &Client,
+        client: Arc<Client>,
```

Comment:
nit: revert 

(@passaro's [change](https://github.com/awslabs/mountpoint-s3/pull/1032/files#diff-ff2b2b24fbf3d7ad046bb857a44d84da70bbdd714b0ea9460f5d1b8a08408effR307) will make Client cloneable, but probably may merge as is and address separately)

Reply:
Is there any reason for the change? I can't see it.

Reply:
We want to use the client in `MemoryLimiter` and wrapping it in `Arc` makes it easier to do that. We were passing the client differently in a few places, for example, we use `Arc<Client>` in the [uploader](https://github.com/awslabs/mountpoint-s3/blob/6cda3049e6b0d627748c16977c97c6e4f6241645/mountpoint-s3/src/upload.rs#L27) but pass it as reference in the part stream. I just have less conflicts in my code if I pass it around as `Arc<Client>`.

Reply:
My conflicting change is now merged. I solved the same problem by requiring `Client: Clone`, which is fine for `S3CrtClient` (while the mock, etc. need to be wrapped in `Arc`).

Reply:
Got it, will rebase and see if it solves the conflict.